spec/lib/declare_schema/field_spec_spec.rb

```diff
@@ -1,69 +1,163 @@
 # frozen_string_literal: true
 
-RSpec.describe 'DeclareSchema Model FieldSpec' do
+begin
+  require 'mysql2'
+rescue LoadError
+end
+
+RSpec.describe DeclareSchema::Model::FieldSpec do
+  let(:model) { double('model', table_options: {}) }
+  let(:col_spec) { double('col_spec', sql_type: 'varchar') }
+
   before do
     load File.expand_path('prepare_testapp.rb', __dir__)
+
+    if Rails::VERSION::MAJOR < 5
+      allow(col_spec).to receive(:type_cast_from_database, &:itself)
+    end
   end
 
-  context 'There are no model columns to change' do
-    it '#different_to should return false for int8 == int8' do
-      subject = DeclareSchema::Model::FieldSpec.new(Object, :price, :integer, limit: 8, null: false, position: 0)
+  describe '#schema_attributes' do
+    describe 'integer 4' do
+      it 'returns schema attributes' do
+        subject = described_class.new(model, :price, :integer, limit: 4, null: false, position: 0)
+        expect(subject.schema_attributes(col_spec)).to eq(type: :integer, limit: 4, null: false)
+      end
+    end
 
-      case Rails::VERSION::MAJOR
-      when 4
-        cast_type = ActiveRecord::Type::Integer.new(limit: 8)
-        col = ActiveRecord::ConnectionAdapters::Column.new("price", nil, cast_type, "integer(8)", false)
-      else
-        sql_type_metadata = ActiveRecord::ConnectionAdapters::SqlTypeMetadata.new(sql_type: "integer(8)", type: :integer, limit: 8)
-        col = ActiveRecord::ConnectionAdapters::Column.new("price", nil, sql_type_metadata, false, "adverts")
+    describe 'integer 8' do
+      it 'returns schema attributes' do
+        subject = described_class.new(model, :price, :integer, limit: 8, null: true, position: 2)
+        expect(subject.schema_attributes(col_spec)).to eq(type: :integer, limit: 8, null: true)
```

Comment:
I wound up using `compact` to throw away `nil` options, so `null: true` winds up appearing...even though it's the Rails defaults. I kind of liked being explicit, since it contradicts our convention of `null: false` in the `fields do` block.

Do you think it's helpful to have the `null: true` appear? It would only be a few lines of code if we'd rather suppress it.

Reply:
I like that `null: true` shows up in the migration. It makes it very clear to the user what the basic contract is and what will be set rather than relying on the user have in depth knowledge of mysql defaults.

Reply:
Great, I'll leave it then.